includes/route.php

```diff
@@ -247,6 +247,9 @@ public function events_details( string $event_slug ) {
 		if ( ! $event ) {
 			$this->die_with_404();
 		}
+		if ( 'draft' === $event->post_status && intval( $event->post_author ) !== $user->ID ) {
```

Comment:
Would this also work but be would reach further?
```suggestion
		if ( 'published' !== $event->post_status && intval( $event->post_author ) !== $user->ID ) {
```

Reply:
But I think we also need to allow admins to view/edit it or people with higher priviledges